sca/src/main/kotlin/factory/StaticCodeAnalyzerFactory.kt

```diff
@@ -0,0 +1,17 @@
+package org.example.factory
+
+import org.example.provider.StaticCodeAnalyzerConfigurer
+import org.example.provider.PrintLnArgumentNonExpressionRuleConfigurer
+import org.example.provider.VariableNamingRuleConfigurer
+
+class StaticCodeAnalyzerProviderFactory {
+
+    private val ruleConfigurers = mapOf(
+        "variableNamingRule" to VariableNamingRuleConfigurer(),
+        "printLnArgumentNonExpressionRule" to PrintLnArgumentNonExpressionRuleConfigurer()
```

Comment:
Just to get it clear , we are not considering the case in which these rules are not required right ?